parrot/src/pfs_service_local.cc

```diff
@@ -153,25 +149,6 @@ process to sleep and wait for actual input to become ready.
 		result = ::read(fd,data,length);
 		if(result>0) last_offset = offset+result;
 
-		if(result==0 && offset==0) {
```

Comment:
Right, named pipes are no longer opened as Parrot files. Instead, they are opened as native file descriptors. That logic is handled here:

https://github.com/cooperative-computing-lab/cctools/blob/89b41ac35584347a24baa4c3a42be29e184309f0/parrot/src/pfs_service_local.cc#L103-L110